pkg/kv/kvserver/kvserverpb/proposer_kv.proto

```diff
@@ -259,6 +254,18 @@ message ReplicatedEvalResult {
   // is applied on the new leaseholder through a Raft snapshot.
   kv.kvserver.readsummary.ReadSummary prior_read_summary = 22;
 
+  message LinkExternalSSTable {
+    roachpb.Span span = 1 [(gogoproto.nullable) = false];
+    string remote_file_loc = 2;
+    string remote_file_path = 3;
+    uint64 backing_file_size = 4;
```

Comment:
we no longer use BackingFileSize. can we stop plumbing this?

Reply:
I think we want it; it can avoid a stat call during open if we have it.

Reply:
I don't think we currently use it though. Are you saying we'd use it here? https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/backupccl/restore_online.go#L300

Reply:
We _set_ it here: https://github.com/cockroachdb/cockroach/blob/6681efbbd7d49f09167febc9ef8db01b191a77ab/pkg/ccl/backupccl/restore_online.go#L296

I'm saying we can use it in pebble, when we open the SST, to avoid needing to stat it to know how far back to open at to find the footer.

Reply:
I'd just leave it for now, and keep this PR a pure code-move to split the RPC and raft proposal paths out of AddSSTable with no behavior change.